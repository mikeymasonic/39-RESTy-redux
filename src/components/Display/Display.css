.Display {
  margin: auto;
  width: 50rem;
  background-color: purple;
  color: whitesmoke;
}
