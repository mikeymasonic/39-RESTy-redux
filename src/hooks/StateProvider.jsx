import React, { createContext, useReducer, useContext } from 'react';
import PropTypes from 'prop-types';

const StateContext = createContext();

const initialState = {
  url: '',
  method: 'GET',
  body: '',
  disable: true,
  headers: {},
  response: {},
  requests: []
};

export function reducer(state, action) {
  switch(action.type) {
    case 'SET_URL':
      return { ...state, url: action.payload };
    case 'SET_METHOD':
      return { ...state, method: action.payload };
    case 'SET_BODY':
      return { ...state, body: action.payload };
    case 'SET_REQUESTS':
      return { ...state, requests: action.payload };
    case 'SET_DISABLE':
      return { ...state, disable: action.payload };
    case 'SET_RESPONSE':
      return { ...state, response: action.payload };
    case 'SET_HEADERS':
      return { ...state, headers: action.payload };
    
    default:
      return state;
  }
}

export const StateProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <StateContext.Provider value={{ state, dispatch }}>
      {children}
    </StateContext.Provider>
  );
};

StateProvider.propTypes = {
  children: PropTypes.node
};

export const useGlobalState = () => {
  const { state } = useContext(StateContext);
  return state;
};

export const useURL = () => {
  const { url } = useGlobalState();
  return url;
};

export const useBody = () => {
  const { body } = useGlobalState();
  return body;
};

export const useMethod = () => {
  const { method } = useGlobalState();
  return method;
};

export const useDispatch = () => {
  const { dispatch } = useContext(StateContext);
  return dispatch;
};

export const useDisable = () => {
  const { disable } = useGlobalState();
  return disable;
};

export const useResponse = () => {
  const { response } = useGlobalState();
  return response;
};

export const useHeaders = () => {
  const { headers } = useGlobalState();
  return headers;
};

export const useRequests = () => {
  const { requests } = useGlobalState();
  return requests;
};
