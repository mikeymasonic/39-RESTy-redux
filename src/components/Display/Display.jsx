import React from 'react';
import PropTypes from 'prop-types';
import ReactJson from 'react-json-view';
import styles from './Display.css';
import { useResponse, useHeaders } from '../../hooks/StateProvider';

const Display = () => {
  const response = useResponse();
  const headers = useHeaders();

  return (
    <div className={styles.Display}>
      <ReactJson src={headers} theme="monokai" name="Headers"/>
      <ReactJson src={response} theme="monokai" name="Response" />
    </div>
  );
};

Display.propTypes = {
  headers: PropTypes.object,
  response: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array
  ]),
};

export default Display;
