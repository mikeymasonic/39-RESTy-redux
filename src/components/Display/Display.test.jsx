import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';
import { StateProvider } from '../../hooks/StateProvider';

describe('Display component', () => {
  it('renders Display', () => {
    const wrapper = shallow(
      <StateProvider>
        <Display />
      </StateProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
