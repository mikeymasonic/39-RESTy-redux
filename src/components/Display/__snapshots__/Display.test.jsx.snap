// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders Display 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "body": "",
        "disable": true,
        "headers": Object {},
        "method": "GET",
        "requests": Array [],
        "response": Object {},
        "url": "",
      },
    }
  }
>
  <Display />
</ContextProvider>
`;
